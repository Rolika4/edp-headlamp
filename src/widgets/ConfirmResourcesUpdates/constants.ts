export const CONFIRM_RESOURCES_UPDATES_DIALOG_NAME = 'CONFIRM_RESOURCES_UPDATES';
