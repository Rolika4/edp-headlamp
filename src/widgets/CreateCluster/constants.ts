export const CREATE_CLUSTER_DIALOG_NAME = 'CREATE_CLUSTER_DIALOG';
