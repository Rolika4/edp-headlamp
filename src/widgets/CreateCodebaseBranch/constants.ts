export const CREATE_CODEBASE_BRANCH_DIALOG_NAME = 'CREATE_CODEBASE_BRANCH';
export const RELEASE_BRANCH_POSTFIX = 'RC';
