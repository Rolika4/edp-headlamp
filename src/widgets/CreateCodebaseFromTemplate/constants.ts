export const CREATE_CODEBASE_FROM_TEMPLATE_DIALOG_NAME = 'CREATE_CODEBASE_FROM_TEMPLATE';
