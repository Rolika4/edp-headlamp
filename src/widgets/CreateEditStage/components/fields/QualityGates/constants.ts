export const DEFAULT_QUALITY_GATE = {
    qualityGateType: 'manual',
    stepName: '',
    autotestName: null,
    branchName: null,
};
