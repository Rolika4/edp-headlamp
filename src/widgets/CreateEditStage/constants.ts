export const CREATE_EDIT_STAGE_DIALOG_NAME = 'CREATE_EDIT_STAGE_DIALOG';
