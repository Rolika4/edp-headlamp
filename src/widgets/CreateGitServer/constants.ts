export const CREATE_GIT_SERVER_DIALOG_NAME = 'CREATE_GIT_SERVER';
