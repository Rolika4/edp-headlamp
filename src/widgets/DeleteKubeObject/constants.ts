export const DELETE_KUBE_OBJECT_DIALOG_NAME = 'DELETE_KUBE_OBJECT';
