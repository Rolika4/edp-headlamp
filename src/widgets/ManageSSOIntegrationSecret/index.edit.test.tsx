/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestWrapper } from '../../../mocks/wrappers/default';
import { INTEGRATION_SECRET_NAMES } from '../../k8s/Secret/constants';
import { ManageSSOIntegrationSecret } from './index';

test('renders ManageSSOIntegrationSecret Edit component', () => {
    render(
        <TestWrapper>
            <ManageSSOIntegrationSecret
                formData={{
                    currentElement: {
                        // @ts-ignore
                        metadata: {
                            name: INTEGRATION_SECRET_NAMES.SSO,
                            uid: 'test-uid',
                            labels: {
                                'app.edp.epam.com/secret-type': 'keycloak',
                            },
                            ownerReferences: [
                                {
                                    apiVersion: 'apiVersion',
                                    kind: 'ExternalSecret',
                                    name: INTEGRATION_SECRET_NAMES.SSO,
                                    uid: 'test-uid',
                                    controller: true,
                                    blockOwnerDeletion: true,
                                },
                            ],
                        },
                        immutable: false,
                        data: {
                            password: 'dGVzdC1wYXNzd29yZA==',
                            username: 'dGVzdC11c2VybmFtZQ==',
                        },
                        type: 'Opaque',
                    },
                    handleDeleteRow: jest.fn(),
                }}
            />
        </TestWrapper>
    );

    const dialog = screen.getByTestId('form');
    expect(dialog).toMatchSnapshot();
});
