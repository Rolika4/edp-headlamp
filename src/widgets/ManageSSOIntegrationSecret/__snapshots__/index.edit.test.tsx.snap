// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ManageSSOIntegrationSecret Edit component 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  data-testid="form"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            style="display: flex;"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <span
                  class="makeStyles-labelWrap-1"
                >
                  <span
                    class="MuiTypography-root makeStyles-label-2 MuiTypography-body1"
                  >
                    User
                  </span>
                  <span />
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                style="display: flex; align-items: flex-end;"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        name="username"
                        placeholder="Enter user name"
                        type="text"
                        value="test-username"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            style="display: flex;"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <span
                  class="makeStyles-labelWrap-1"
                >
                  <span
                    class="MuiTypography-root makeStyles-label-2 MuiTypography-body1"
                  >
                    Password
                  </span>
                  <span />
                </span>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                style="display: flex; align-items: flex-end;"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                        name="password"
                        placeholder="Enter password"
                        type="password"
                        value="test-password"
                      />
                      <div
                        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                      >
                        <button
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <span />
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-content-xs-space-between"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <span />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                undo changes
              </span>
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
